src/mss/parser/MssParser.js

```diff
@@ -459,52 +460,65 @@ function MssParser(config) {
 
 
     function createPRContentProtection(protectionHeader) {
-        const keySystems = protectionController ? protectionController.getKeySystems() : null;
-        let ksPlayReady;
-
-        for (let i = 0; i < keySystems.length; i++) {
-            if (keySystems[i].systemString && keySystems[i].systemString.indexOf('playready') !== -1) {
-                ksPlayReady = keySystems[i];
-                break;
-            }
-        }
-
-        let contentProtection = {};
-        let pro;
-
-        pro = {
+        let pro = {
             __text: protectionHeader.firstChild.data,
             __prefix: 'mspr'
         };
-
-        if (ksPlayReady) {
-            contentProtection.schemeIdUri = ksPlayReady.schemeIdURI;
-            contentProtection.value = ksPlayReady.systemString;
-            contentProtection.pro = pro;
-            contentProtection.pro_asArray = pro;
-        }
-
-        return contentProtection;
+        return {
+            schemeIdUri: 'urn:uuid:9a04f079-9840-4286-ab92-e65be0885f95',
+            value: 'com.microsoft.playready',
+            pro: pro,
+            pro_asArray: pro
+        };
     }
 
-    function createWidevineContentProtection(/*protectionHeader*/) {
-        const keySystems = protectionController ? protectionController.getKeySystems() : null;
-        let ksWidevine;
+    function createWidevineContentProtection(protectionHeader, KID) {
+        // Create Widevine CENC header and encode it (Protocol Buffer)
+        let wvCencHeader = new WidevineCencHeader().encode({
+            algorithm: 1,
+            keyId: [KID],
+            trackTypeDeprecated: 'SD'
+        }).finish();
+
+        // Create a pssh box
+        let length = 12 /* box length, type, version and flags */ + 16 /* SystemID */ + 4 /* data length */ + wvCencHeader.length;
+        let pssh = new Uint8Array(length);
+        let i = 0;
 
-        for (let i = 0; i < keySystems.length; i++) {
-            if (keySystems[i].systemString && keySystems[i].systemString.indexOf('widevine') !== -1) {
-                ksWidevine = keySystems[i];
-                break;
+        // Set box length value
+        pssh[i++] = (length & 0xFF000000) >> 32;
```

Comment:
Should this probably be '>>> 24;' ? Move three bytes left and fill with zeros?

Reply:
Yes of course! Thanks for pointing out. I will correct.